private/buf/bufcli/errors.go

```diff
@@ -164,19 +169,29 @@ func NewTemplateNotFoundError(owner string, name string) error {
 // Note that this function will wrap the error so that the underlying error
 // can be recovered via 'errors.Is'.
 func wrapError(err error) error {
-	if err == nil || (err.Error() == "" && !rpc.IsError(err)) {
-		// If the error is nil or empty and not an rpc error, we return it as-is.
-		// This is especially relevant for commands like lint and breaking.
+	// Attempt to convert err to a Connect error.  If it cannot be converted, we return it as-is.
+	// This is especially relevant for commands like lint and breaking.
+	connectError, ok := asConnectError(err)
+	if !ok {
 		return err
 	}
-	rpcCode := rpc.GetErrorCode(err)
+
+	connectCode := connectError.Code()
+	connectMessage := connectError.Message()
 	switch {
-	case rpcCode == rpc.ErrorCodeUnauthenticated, isEmptyUnknownError(err):
+	case connectCode == connect.CodeUnauthenticated, isEmptyUnknownError(err):
 		return errors.New(`Failure: you are not authenticated. Create a new entry in your netrc, using a Buf API Key as the password. For details, visit https://docs.buf.build/bsr/authentication`)
-	case rpcCode == rpc.ErrorCodeUnavailable:
+	case connectCode == connect.CodeUnavailable:
 		return fmt.Errorf(`Failure: the server hosted at that remote is unavailable: %w`, err)
 	}
-	return fmt.Errorf("Failure: %w", err)
+	return fmt.Errorf("Failure: %s", connectMessage)
+}
+
+// asConnectError uses errors.As to unwrap any error and look for a connect *Error.
+func asConnectError(err error) (*connect.Error, bool) {
```

Comment:
I'd love to see this helper as part of the core connect library, something we've floated in the past:

```
CheckAs(err) (*connect.Error, bool)
```

It's easy enough to write (as you did), and in core we inline it:

```
if connectErr := (&connect.Error{}); errors.As(err, &connectErr) {
    // handle connect error
}
```

Having one way to do this as defined by the connect library would be ideal. Nbd, but a nicer developer experience I think.

Reply:
There actually already is a function like this in the library, it's just private.  I'm definitely in favor of making it exported:

https://github.com/bufbuild/connect-go/blob/main/error.go#L134

Reply:
It used to be exported! I have no objection to exporting it again if it's useful.